Performance.md

```diff
@@ -0,0 +1,290 @@
+For that reason, TypeScript provides the `isolatedModules` flag to warn when using them.
+Using `isolatedModules` means that your codebase is safe for tools to use TypeScript APIs like `transpileModule` or alternative compilers like Babel.
+
+As an example of this in action, ts-loader provides
+
+* the `transpileOnly` flag to use `transpileModule`.
+* the `useBabel` flag
+
+<!-- TODO -->
+
+# Investigating Issues
+
+There are certain ways to get hints of what might be going wrong.
+
+## `extendedDiagnostics`
+
+You can run TypeScript with `--extendedDiagnostics` to get a printout of where the compiler is spending its time.
+
+```
+Files:                         6
+Lines:                     24906
+Nodes:                    112200
+Identifiers:               41097
+Symbols:                   27972
+Types:                      8298
+Memory used:              77984K
+Assignability cache size:  33123
+Identity cache size:           2
+Subtype cache size:            0
+I/O Read time:             0.01s
+Parse time:                0.44s
+Program time:              0.45s
+Bind time:                 0.21s
+Check time:                1.07s
+transformTime time:        0.01s
+commentTime time:          0.00s
+I/O Write time:            0.00s
+printTime time:            0.01s
+Emit time:                 0.01s
+Total time:                1.75s
+```
+
+The most relevant information for most users is:
+
+Field   | Meaning
+--------|---------
+`Files` | the number of files that the program is including (use `listFiles` to see what they are).
+`I/O Read time` | time spent reading from the file system - this includes traversing `include`'d folders.
+`Parse time` | time spent scanning and parsing the program
+`Program time` | combined time spent performing reading from the file system, scanning and parsing the program, and other calculation of the program graph. These steps are intermingled and combined here because files need to be resolved and loaded once they're included via `import`s and `export`s.
+`Bind time` | Time spent building up various semantic information that is local to a single file.
+`Check time` | Time spent type-checking the program.
+`transformTime time` | Time spent rewriting TypeScript ASTs (trees that represent source files) into forms that work in older runtimes.
+`commentTime` | Time spent calculating comments in output files.
+`I/O Write time` | Time spent writing/updating files on disk.
+`printTime` | Time spent calculating the string representation of an output file.
+
+<!-- TODO: is this important? Looks identical to printTime
+
+`Emit time` | ...
+
+-->
+
+## `showConfig`
+
+It's not always obvious what settings a compilation is being run with when running `tsc`, especially given that `tsconfig.json`s can extend other configuration files.
+`showConfig` can explain what `tsc` will calculate for an invocation.
+
+```sh
```

Comment:
Note that the output will look like a tsconfig file but can't necessarily be used as one, since it may include command-line only options.